### DISCUSSION TITLE
NATS websocket

### DISCUSSION THREAD

**Discussion by ahmaddxb** [amidaware/tacticalrmm#1272] [General]
Is there a check I can do to see if NATS websocket connections are working from the remote machines? 

I'm using Traefik proxy and I want to make sure I have set it up correctly.

**Comment by dinger1986** [comment]
converted to a discussion.

Do you think they arent working? Are your Agents connecting? Is port 4222 opened?

**Reply by ahmaddxb** [reply]
My agents are connected. Port 4222 is still open, I have not closed it yet.

**Reply by ahmaddxb** [reply]
what would I notice is not working if NATS websocket connection is broken?

**Reply by dinger1986** [reply]
I am pretty sure that once it converts them to using websockets it doesnt use the port anymore and only uses the port if thats manually enabled. 

Easiest way to test is close the port, then run send command and send `ping 8.8.8.8` if that works then it will confirm websockets is in place and working correctly, which I am sure it will, maybe test it on a handful for peace of mind (make sure the ones you are testing on are agent v2.3.0) then open the port again just incase theres any old agents which haven't updated.

**Reply by ahmaddxb** [reply]
ok thanks for your help. I found out how to break it from working. So now I can go from here to make sure I set up NATS websocket in Traefik correctly

**Reply by dinger1986** [reply]
Perfect, was it miles off if what I said? Maybe worth writing up how you tested it etc for others. Thanks

**Reply by ahmaddxb** [reply]
What you said was good, I found out to test a nats connection is working is as simple as send a ping command and getting a reply. 

I found out the websocket config I did made no difference as nats websocket is still worked using the config to allow 4222 found here https://gitlab.com/NiceGuyIT/tactical-goodies/-/blob/main/traefik/conf.d/routers-api-nats.a8n.tools.toml

Once I disabled that config above things broke.

Not sure what is the correct config to use for nats websocket and Traefik but even if i dont figure it out it seem to continue to work as long as the above config is still active.

**Reply by dinger1986** [reply]
Maybe worth asking on discord probably someone there worth speaking to.
